Java/Mohican/Mohican.java

```diff
@@ -0,0 +1,44 @@
+package com.gymfox.statics.mohicanclass;
+
+public class Mohican {
+    private static Mohican lastMohican = null;
+    private static Mohican prev;
+    private String name;
+
+    public Mohican(String name) {
+        this.name = name;
+
+        prev = lastMohican;
+        lastMohican = this;
+    }
+
+    public static void deleteLastMohican() throws NoLastMohicanException {
+        validate();
+
+        lastMohican = prev;
```

Comment:
Looks like after deleting fields `lastMohican` and `prev` will refer to same object. After the first delete the same mohican will be deleted each time. Seems like `prev` shouldn't be static.

Pretty serious bug, should write better tests.